xray/core/indexing.py

```diff
@@ -368,7 +368,12 @@ class PandasIndexAdapter(utils.NDArrayMixin):
     def __init__(self, array, dtype=None):
         self.array = utils.safe_cast_to_index(array)
         if dtype is None:
-            dtype = array.dtype
+            # if the index -'hides' its dtype such as
+            # PeriodIndex use an object dtype
+            if self.array._box_scalars:
```

Comment:
Yes; how do you balance that against the 'generalizable-ness' of this - will this then fail for `IntervalIndex` when that comes along, etc?